presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSourceFactory.java

```diff
@@ -0,0 +1,321 @@
+                    orcReaderOptions,
+                    hiveFileContext.isCacheable(),
+                    dwrfEncryptionProvider,
+                    dwrfKeyProvider,
+                    hiveFileContext.getStats());
+
+            List<HiveColumnHandle> physicalColumns = getPhysicalHiveColumnHandles(columns, useOrcColumnNames, reader.getTypes(), path);
+            ImmutableMap.Builder<Integer, Type> includedColumns = ImmutableMap.builder();
+            ImmutableList.Builder<ColumnReference<HiveColumnHandle>> columnReferences = ImmutableList.builder();
+            for (HiveColumnHandle column : physicalColumns) {
+                if (column.getColumnType() == REGULAR) {
+                    Type type = typeManager.getType(column.getTypeSignature());
+                    includedColumns.put(column.getHiveColumnIndex(), type);
+                    columnReferences.add(new ColumnReference<>(column, column.getHiveColumnIndex(), type));
+                }
+            }
+
+            if (!physicalColumns.isEmpty() && physicalColumns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
+                return new AggregatedOrcPageSource(physicalColumns, reader.getFooter(), typeManager, functionResolution);
+            }
+
+            OrcPredicate predicate = new TupleDomainOrcPredicate<>(effectivePredicate, columnReferences.build(), orcBloomFiltersEnabled, Optional.of(domainCompactionThreshold));
+
+            OrcBatchRecordReader recordReader = reader.createBatchRecordReader(
+                    includedColumns.build(),
+                    predicate,
+                    start,
+                    length,
+                    hiveStorageTimeZone,
+                    systemMemoryUsage,
+                    INITIAL_BATCH_SIZE);
+
+            return new OrcBatchPageSource(
+                    recordReader,
+                    reader.getOrcDataSource(),
+                    physicalColumns,
+                    typeManager,
+                    systemMemoryUsage,
+                    stats,
+                    hiveFileContext.getStats());
+        }
+        catch (Exception e) {
+            try {
+                orcDataSource.close();
+            }
+            catch (IOException ignored) {
+            }
+            if (e instanceof PrestoException) {
+                throw (PrestoException) e;
+            }
+            String message = splitError(e, path, start, length);
+            if (e.getClass().getSimpleName().equals("BlockMissingException")) {
+                throw new PrestoException(HIVE_MISSING_DATA, message, e);
+            }
+            throw new PrestoException(HIVE_CANNOT_OPEN_SPLIT, message, e);
+        }
+    }
+
+    private static String splitError(Throwable t, Path path, long start, long length)
+    {
+        return format("Error opening Hive split %s (offset=%s, length=%s): %s", path, start, length, t.getMessage());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: start (declared at line 317), length (declared at line 317). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 319. Consider formatting this data with &#39;%d&#39; instead.